### PR TITLE
publishing-bot: Use go1.15.0 as default version and go1.13.15 for previous releases

### LINKED ISSUE #1474: [golang] go1.14.7 and go1.13.15 security updates
**What would you like to be added:**

Tracking issue for go1.14.7 and go1.13.15 security updates
Announcement: https://groups.google.com/d/topic/golang-announce/NyPIaucMgXo/discussion

**Work items:**

- [x] kube-cross bump: https://github.com/kubernetes/release/pull/1473
- [x] image promotion: https://github.com/kubernetes/k8s.io/pull/1119
- [x] k/release/k8s-cloud-builder bump: https://github.com/kubernetes/release/pull/1475
- [x] k/repo-infra bump: https://github.com/kubernetes/repo-infra/pull/205
- [x] k/repo-infra release cut: https://github.com/kubernetes/repo-infra/releases/tag/v0.0.11
- [x] kubekins/krte variants bump: https://github.com/kubernetes/test-infra/pull/18758
- [ ] publishing-bot bump: https://github.com/kubernetes/kubernetes/pull/93998

### go1.14.7

N/A -- `master` and `release-1.19` are [moving towards go1.15](https://github.com/kubernetes/release/issues/1421)

### go1.13.15

- [x] k/k bump (`release-1.18`): https://github.com/kubernetes/kubernetes/pull/93953
  - [x] k/k bazel rules_go: https://github.com/kubernetes/kubernetes/pull/93953

### Cherry picks

- [x] 1.17: https://github.com/kubernetes/kubernetes/pull/93955
- [x] 1.16: https://github.com/kubernetes/kubernetes/pull/93956

/assign
/area dependency
cc: @kubernetes/release-engineering

### PR DESCRIPTION
**Pull request by justaugustus**
**What type of PR is this?**

/kind cleanup
/area dependency

**What this PR does / why we need it**:

- publishing-bot: Use go1.15.0 as default version
- publishing-bot: Use go1.13.15 for 1.18, 1.17, and 1.16 release branches

Tracking issues: https://github.com/kubernetes/release/issues/1421, https://github.com/kubernetes/release/issues/1474
Closes: https://github.com/kubernetes/release/issues/1474

/assign @nikhita @liggitt
cc: @kubernetes/release-engineering @kubernetes/publishing-bot-reviewers 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/publishing/rules.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ recursive-delete-patterns:
 - BUILD.bazel
 - "*/BUILD.bazel"
 - Gopkg.toml
-default-go-version: 1.14.6
+default-go-version: 1.15.0
```
Comment (nikhita) [inline staging/publishing/rules.yaml:7]:
```suggestion
default-go-version: 1.15
```

Comment (nikhita) [reply staging/publishing/rules.yaml:7]: Would need to do this for all 1.15 versions below

Comment (justaugustus) [reply staging/publishing/rules.yaml:7]: Oh geez, really?
_shakes fist at go minor versions_

Comment (justaugustus) [reply staging/publishing/rules.yaml]: Fixed.

File: staging/publishing/rules.yaml [hunk 2]
```diff
@@ -1640,9 +1640,9 @@ rules:
       branch: master
       dir: staging/src/k8s.io/controller-manager
     name: master
-    go: 1.13.9
+    go: 1.13.15
```
Comment (nikhita) [inline staging/publishing/rules.yaml:1643]: master would be using go 1.15 so we'd need to remove this line (was incorrect before)

Comment (justaugustus) [reply staging/publishing/rules.yaml]: Fixed.

**Comment by nikhita** [comment]
Thanks, Stephen!

/lgtm
/approve

**Pull request merged**